testing/mocker/create_derivatives.py

```diff
@@ -0,0 +1,230 @@
+        list_modalities = ["T2w"]
+        CreateBIDSDerivatives(
+            id,
+            index_subject,
+            path_to_mock_data,
+            list_sessions,
+            list_modalities,
+        )
+
+    def run(self):
+        """
+        Main entry point of the multiprocess.
+        :return:
+        """
+
+        # Create the sessions
+        self.create_sessions(self.list_sessions, self.list_bids_details)
+
+        # Save mock_json dictionary into a json file:
+        mock_json: dict = self.generate_dataset_description_dict()
+
+        path_json_file: Path = (
+            self.path_to_mock_data / "derivatives" / "dataset_description.json"
+        )
+        with open(str(path_json_file), "w") as f:
+            json.dump(mock_json, f, indent=4, sort_keys=True)
+
+    def create_sessions(
+        self, sessions: List[str] or None, list_bids_details: List[dict]
+    ):
+        """
+        Within each session, create the expected nifti label files
+        :param sessions:
+        :param list_bids_details:
+        :return:
+        """
+        if sessions:
+            for session in sessions:
+                self.create_session_specific_labels(session, list_bids_details)
+        else:
+            self.create_session_specific_labels(None, list_bids_details)
+
+    def create_session_specific_labels(
+        self, session: str or None, list_bids_details_in_lists: List[dict]
+    ):
+        """
+        Create the expected nifti files for a given session.
+        :param session:
+        :param list_bids_details:
+        :return:
+        """
+        # Take a single bids_deetail dict from the list.
+        # This dictionary contain LISTS
+        for a_bids_detail_dict in list_bids_details_in_lists:
+
+            # Break down a single bids_detail_dictionary into concretized individual specific bids parameter pairs.
+            # This is because bids_details could have list PER modalities.
+            # e.g. acq = ["T1w", "T2w", "FLAIR", "PD"] needs to be broken down into FOR separate calls.
+            # Source Inspiration: https://stackoverflow.com/a/61335465
+
+            bids_standard_keys, bids_values = zip(*a_bids_detail_dict.items())
+
+            list_permuted_dicts: List[dict] = [
+                dict(zip(bids_standard_keys, a_set_of_permuted_values))
+                for a_set_of_permuted_values in itertools.product(*bids_values)
+            ]
+
+            for a_permuted_bids_detail_dict in list_permuted_dicts:
+                self.create_one_json_nii_pair(session, a_permuted_bids_detail_dict)
+
+    def create_one_json_nii_pair(self, session: str or None, bids_detail: dict):
+        """
+        Create a nifti/JSON image pairs with an explicitly set list of session and model
+        :param session:
+        :param contrast:
+        :param kwargs:
+        :return:
+        """
+
+        file_stem: str = self.create_file_stem(session, bids_detail)
+
+        self.generate_a_json_file(
```

Comment:
This will create the exact same JSON file for the label than for the raw image.
This is not common in BIDS and when a derivative has a JSON, it will _generally_ contain information specific to the derivative itself (AFAIK).

So maybe we could have different JSON for images vs. labels.

We could have these fields for examples:
[Description](https://bids-specification.readthedocs.io/en/stable/05-derivatives/02-common-data-types.html#common-file-level-metadata-fields) from here in BIDS.
[Author and Date](https://github.com/ivadomed/ivadomed/wiki/repositories#derivatives) from our own wiki 😅.